### DISCUSSION TITLE
Enable dependabot and code scanning (codeql) with code

### DISCUSSION THREAD

**Discussion by ceelinab** [community/community#53755] [Programming Help] [answered]
### Body

Hello, I am currently working on my bachelors thesis in digital infrastructure and cyber security and would like to write some code that can enable dependabot and code scanning in github. Is there any way to do this - in terraform perhaps? Or do I have to do this manually? 

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by hmakholm** [comment]
I don't know an answer for Dependabot, but would https://github.blog/changelog/2023-03-21-code-scanning-api-to-enable-default-setup-with-codeql-on-a-repository/ be what you're looking for with CodeQL?

**Reply by ceelinab** [reply]
I'm just wondering if there is a way to simply enable codeql with any form of code, instead of going in manually in github repository and click the enable button :D

**Reply by hmakholm** [reply, marked as answer]
Unless I misunderstand you, that's exactly what the API announced in that blog post describes.

**Reply by ceelinab** [reply]
Thank you for the help. The API is exactly what i need.
